doc/source/serve/autoscaling-guide.md

```diff
@@ -0,0 +1,369 @@
+By default, `num_replicas` is 1.
+
+```{literalinclude} doc_code/managing_deployments.py
+:start-after: __scaling_out_start__
+:end-before: __scaling_out_end__
+:language: python
+```
+
+## Autoscaling Important Concepts
+
+Instead of setting a fixed number of replicas for a deployment, you can configure a deployment to autoscale based on incoming traffic. The Serve autoscaler adjusts to traffic spikes by observing queue sizes and making scaling decisions to add or remove replicas. To configure it, you can set the `autoscaling_config` field in deployment options.
+
+:::{note}
+The Ray Serve Autoscaler is an application-level autoscaler that sits on top of the [Ray Autoscaler](cluster-index).
+Concretely, this means that the Ray Serve autoscaler asks Ray to start a number of replica actors based on the request demand.
+If the Ray Autoscaler determines there aren't enough available CPUs to place these actors, it responds by requesting more Ray nodes.
+The underlying cloud provider will then respond by adding more nodes.
+Similarly, when Ray Serve scales down and terminates some replica actors, it will try to do so in a way that results in the most nodes having no Ray actors or tasks running on them, at which point the Ray autoscaler will remove those nodes.
+:::
+
+To learn about the architecture underlying Ray Serve Autoscaling, see {ref}`serve-autoscaling-architecture`.
+
+Before diving into an example, we will go over some foundational concepts that Serve autoscaling is built on top of.
+
+### Define the Steady State of your System
+
+To define what the steady state of your deployments should be like, you need to set values for `target_num_ongoing_requests_per_replica` and `max_concurrent_queries`.
+
+Serve will scale the number of replicas for a deployment up or down based on the average number of ongoing requests per replica. More specifically, Serve will compare the *actual* average number of ongoing requests per replica with the target value you set in the autoscaling config and make upscale or downscale decisions from that. The target value is set by `target_num_ongoing_requests_per_replica`, and Serve will try to make sure that each replica has roughly that number
+of requests being processed and waiting in the queue. We recommend you benchmark your application code and set this number based on end to end latency objective.  There is also a maximum limit for the queue that is respected by proxies when assigning requests to replicas, which is defined by `max_concurrent_queries`.
+
+
+:::{note}
+- It is always recommended to load test your workloads. For example, if the use case is latency sensitive, you can lower the `target_num_ongoing_requests_per_replica` number to maintain high performance.
+- Internally, the autoscaler will decide to scale up or down by comparing `target_num_ongoing_requests_per_replica` to the number of `RUNNING` and `PENDING` tasks on each replica.
```

Comment:
```suggestion
- Internally, the autoscaler will decide to scale up or down by comparing `target_num_ongoing_requests_per_replica` to the number of requests running and pending on each replica.
```

`RUNNING` and `PENDING` are not publicly-exposed states so this may be confusing